src/execution/physical_plan/plan_comparison_join.cpp

```diff
@@ -29,6 +31,51 @@ static bool CanPlanIndexJoin(Transaction &transaction, TableScanBindData *bind_d
 	return true;
 }
 
+void CheckForPerfectJoinOpt(LogicalComparisonJoin &op, PerfectHashJoinStats &join_state) {
+	// we only do this optimization for inner joins
+	if (op.join_type != JoinType::INNER)
+		return;
+	// with one condition
+	if (op.conditions.size() != 1)
+		return;
+	// with propagated statistics
+	if (op.join_stats.size() <= 0)
+		return;
+	// with equality condition
+	for (auto &&condition : op.conditions) {
+		if (condition.comparison != ExpressionType::COMPARE_EQUAL) {
+			return;
+		}
+	}
+	// with integral types
+	for (auto &&join_stat : op.join_stats) {
+		if (!join_stat->type.IsIntegral() || join_stat->type == LogicalTypeId::HUGEINT) {
```

Comment:
Can we check physical types here instead? Then we can also do perfect hash joins for dates/times/timestamps.